File: dart/acronym/lib/acronym.dart
class Acronym {
  String abbreviate(String name){
    RegExp punct = new RegExp(r"[^A-Za-z']+");
    String accro = name.toUpperCase().split(punct).fold('', (String a, String b) => (a + b[0]));
    return accro;
  }
}

File: csharp/acronym/Acronym.cs
using System;
using System.Linq;

public static class Acronym
{
    public static string Abbreviate(string phrase)
    {
        string acronym = String.Empty;
        string[] words = phrase.ToUpper().Replace("-", " ").Replace("_", " ").Split(" ");
        foreach (var word in words)
        {
            if (!String.IsNullOrWhiteSpace(word))
            {
                acronym += word[0];
            }
        }

        return acronym;
    }
}